### DISCUSSION TITLE
SharePoint Or Microsoft Search, do I need to change to Graph?

### DISCUSSION THREAD

**Discussion by wideserg** [microsoft-search/pnp-modern-search#3123] [General]
Hi all,

I'm building intranet portal and using PnP Search solution as well. Classic Search works pretty well for me right now, but I want to know if there will be any massive benefit if Iswitch to Microsoft Search, didn`t find any info about than except that it is more modern way.

Anyone can comment? Any use-cases where classic search sucks?

Thank you!

**Comment by FranckyC** [comment]
@wideserg well, as long as you target only SharePoint content, classic search is still an option. However. Microsoft Search becomes interesting when you need to integrate with other sources as well (LOB applications or in Microsoft 365, like messages, Teams, etc.). If your solution needs to be more future proof, MS Search is more an option in my opinion. Unfortunately, the Microsoft Search data source in the PnP Modern Search solution is not up to date regarding the APi capabilities, but for basic use cases that is enough.

**Reply by kasperbolarsen** [reply]
I agree with Franck, and just wishes to add that you can mix and match Result web parts using SharePoint as the source with Result web parts using Microsoft Search as the source,  getting the best of both worlds 😁
